photon/libphoton/photon/photon-tile.hh

```diff
@@ -105,14 +106,30 @@ struct TileEmu {
     UNARY_OP(vsigmoid_u8,  uint8_t, 64, arithmetic::sigmoid_u8)
     UNARY_OP(vsigmoid_fp16, half, 32, arithmetic::sigmoid_fp16)
 
-    inline void vld(uint8_t * addr, VREG_T& dst) {
-        hooks->on_mem_read(tile_id, (uintptr_t)addr);
-        memcpy(dst.bytes, addr, 64);
+    static inline void vld(uint8_t * addr, VREG_T& dst) {
+        assert(64 % BLOCK_SIZE == 0);
+        for (size_t i = 0; i < 64; i += BLOCK_SIZE) {
```

Comment:
We are eventually just reading from *addr anyway... from above comment, just remove the data array entirely from the cache struct and use it just to track tags / presence of data.